### DISCUSSION TITLE
Laravel 11: change bootstrap/app.php withExceptions or withMiddleware from a service provider

### DISCUSSION THREAD

**Discussion by m-lotze** [laravel/framework#50902] [Q&A]
How would you extend the `withExceptions` callback in `bootstrap/app.php` from a service provider, e.g. from a package? 

In Laravel <= 10 you would just extend your local ExceptionHandler from the packaged one. How can you add something like `$exceptions->report(fn (Throwable $e) => Mail::send(new Exception($e)));` to the callback?

**Comment by jakubkubera** [comment]
I had the same problem. I solved it. I fired up my old custom handler in bootstrap\app.php like this:

```
$app->singleton(
    Illuminate\Contracts\Debug\ExceptionHandler::class,
    App\Exceptions\Handler::class
);

$using ??= fn () => true;

$app->afterResolving(
    App\Exceptions\Handler::class,
    fn ($handler) => $using(new Illuminate\Foundation\Configuration\Exceptions($handler)),
);
```

**Comment by ghost** [comment]
Laravel 11 should had simplified things not complicate them...

**Comment by jakubkubera** [comment]
Laravel makes it easier when you have a version 11 application structure rather than older ones.

**Reply by ghost** [reply]
Makes it simpler for projects started from scratch in v11 and while learning laravel starting with v11:)

**Comment by jncalderon** [comment]
I just upgrade v10 to v11. I create a v11 example and replicate that in my project and is working well. I have another [issue](https://github.com/laravel/framework/discussions/51203) but is other story.

**Comment by rthossein** [comment]
Hi
https://laracasts.com/discuss/channels/laravel/custom-error-handler-laravel-11
It might provide some useful insights for your issue. Hope it helps!
